clang/include/clang/AST/ExprCXX.h

```diff
@@ -2388,15 +2389,20 @@ class CXXNewExpr final
 
   /// Whether this new-expression has any initializer at all.
   bool hasInitializer() const {
-    return CXXNewExprBits.StoredInitializationStyle > 0;
+    switch (getInitializationStyle()) {
+    case CXXNewInitializationStyle::None:
+      return true;
+    case CXXNewInitializationStyle::Implicit:
+    case CXXNewInitializationStyle::Call:
+    case CXXNewInitializationStyle::List:
+      return false;
+    }
```

Comment:
```suggestion
    return getInitializationStyle() != CXXNewInitializationStyele::None;
```

(Isn't the logic of that code right now inverted?)

Reply:
Behavioral change wasn't intended and is now fixed (thank you!)
I find inequality checks against enums not very robust, so I refactored this into switch.